docs/Tracing.md

```diff
@@ -0,0 +1,194 @@
+    * `STANDARD` - The default setting for instrumented runs: currently equivalent to `RUNTIME | OPERATOR`. 
+    * `DEBUG` - Additional events for in depth debugging. This is intended for engineers familiar with the device and with Glow to triage performance issues in particular models. Many more events are permitted here, but included events should focus on what is high value for diagnosis issues.
+
+Often the device specific runtime components are capable of emitting a large amount of hardware events. Please keep Trace Event memory size reasonable: a good guideline is about 1-2 Kb per millisecond of execution time on the device.
+
+### The Format
+
+Inside glow TraceEvents are structs that have the following fields (loosely matching the JSON parameters from the TraceEvent format):
+
+```
+struct TraceEvent {
+  string name;
+  uint64_t timestamp;
+  char type;
+  int tid;
+  uint64_t duration;
+};
+```
+
+* name: The human interpretable name for the event. For operator events this is often the Glow-name for the node or instruction.
+* timestamp: The time reported in microseconds since epoch in the std::system_clock domain. Trace events are aggregated from a variety of sources (multiple devices + the host) and it's important that all timestamps are in the same domain so they can be visualized in a single trace. For an example of timestamp conversion see the OpenCLDeviceManager.
+* type: A one character code for the kind of TraceEvent this represents. Glow supports four types of event:
+    * “B” and “E”: where the timestamp represents the begin and end of the event respectively.
+    * “I”: an event which has no duration, only an instant in time at the the timestamp.
+    * “X”: an event where the timestamp represents the begin of the event, and the duration field encodes the length. Complete events with this type can encode both beginning and end in a single event.
+    * More detail on each type of event can be found in Chrome's [Trace Event Format doc](https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU).
+* tid: An identifier for the computation thread or flow that generated this event so that events produced in the same flow are associated in the result trace. This may or may not be equivalent to a system TID, but the Glow Runtime does not guarantee it will use the TID of the thread it logs the event in (and in fact does not use it). In the Chrome trace visualizer each distinct Thread ID will be represented as a different lane (row) in the trace.
+* duration: Used for Complete events (type 'X') to encode the time taken by the event.
+
+### Visualizing and Inspecting Traces
+
+One you have a JSON trace it is simple to visualize it using Chrome:
+
+1. Browse to `chrome://tracing`
+2. Drag and drop a Trace Event file, or hit the Load button:
+3. ![](chrome-tracing.png)
```

Comment:
missing? (it's not displayed for me in the preview mode)

Reply:
I think its just because the images are in the same commit but preview doesn't know to look for them. If you look at it in the branch of my fork it works: https://github.com/nickgg/glow/blob/traceDocs/docs/Tracing.md